datafusion/sqllogictest/test_files/string/string_literal.slt

```diff
@@ -829,3 +838,95 @@ SELECT
     'a' LIKE '%'
 ----
 NULL true true
+
+# \ is an implicit escape character
```

Comment:
When working on https://github.com/apache/datafusion/pull/13260 i didn't know that \ is an implicit escape character. I couldn't find any tests for this, so here they come.